src/darwin/mod.rs

```diff
@@ -400,50 +417,65 @@ impl Font {
+                ptr::null_mut(),
+                rasterized_width as usize,
+                rasterized_height as usize,
+                8, // bits per component
+                rasterized_width as usize * 4,
+                CGColorSpace::new_device_rgb().as_deref(),
+                CGImageAlphaInfo::PremultipliedFirst.0 | CGBitmapInfo::ByteOrder32Host.0,
+            )
+        }
+        .unwrap();
 
         let is_colored = self.is_colored();
 
         // Set background color for graphics context.
         let bg_a = if is_colored { 0.0 } else { 1.0 };
-        cg_context.set_rgb_fill_color(0.0, 0.0, 0.0, bg_a);
+        unsafe { CGContext::set_rgb_fill_color(Some(&cg_context), 0.0, 0.0, 0.0, bg_a) };
 
         let context_rect = CGRect::new(
-            &CGPoint::new(0.0, 0.0),
-            &CGSize::new(f64::from(rasterized_width), f64::from(rasterized_height)),
+            CGPoint::new(0.0, 0.0),
+            CGSize::new(f64::from(rasterized_width), f64::from(rasterized_height)),
         );
 
-        cg_context.fill_rect(context_rect);
+        unsafe { CGContext::fill_rect(Some(&cg_context), context_rect) };
 
-        cg_context.set_allows_font_smoothing(true);
-        cg_context.set_should_smooth_fonts(*FONT_SMOOTHING_ENABLED);
-        cg_context.set_allows_font_subpixel_quantization(true);
-        cg_context.set_should_subpixel_quantize_fonts(true);
-        cg_context.set_allows_font_subpixel_positioning(true);
-        cg_context.set_should_subpixel_position_fonts(true);
-        cg_context.set_allows_antialiasing(true);
-        cg_context.set_should_antialias(true);
+        unsafe {
+            CGContext::set_allows_font_smoothing(Some(&cg_context), true);
+            CGContext::set_should_smooth_fonts(Some(&cg_context), *FONT_SMOOTHING_ENABLED);
+            CGContext::set_allows_font_subpixel_quantization(Some(&cg_context), true);
+            CGContext::set_should_subpixel_quantize_fonts(Some(&cg_context), true);
+            CGContext::set_allows_font_subpixel_positioning(Some(&cg_context), true);
+            CGContext::set_should_subpixel_position_fonts(Some(&cg_context), true);
+            CGContext::set_allows_antialiasing(Some(&cg_context), true);
+            CGContext::set_should_antialias(Some(&cg_context), true);
+        }
 
         // Set fill color to white for drawing the glyph.
-        cg_context.set_rgb_fill_color(1.0, 1.0, 1.0, 1.0);
+        unsafe { CGContext::set_rgb_fill_color(Some(&cg_context), 1.0, 1.0, 1.0, 1.0) };
         let rasterization_origin =
             CGPoint { x: f64::from(-rasterized_left), y: f64::from(rasterized_descent) };
 
-        self.ct_font.draw_glyphs(
-            &[glyph_index as CGGlyph],
-            &[rasterization_origin],
-            cg_context.clone(),
-        );
+        unsafe {
+            self.ct_font.draw_glyphs(
+                NonNull::from(&[glyph_index as CGGlyph][0]),
+                NonNull::from(&[rasterization_origin][0]),
```

Comment:
Again, why are we creating arrays just to immediately access their only element?